storm-core/src/jvm/org/apache/storm/zookeeper/Zookeeper.java

```diff
@@ -0,0 +1,355 @@
+import org.apache.curator.framework.recipes.leader.LeaderLatchListener;
+import org.apache.curator.framework.recipes.leader.Participant;
+import org.apache.curator.framework.state.ConnectionStateListener;
+import org.apache.storm.Config;
+import org.apache.storm.callback.DefaultWatcherCallBack;
+import org.apache.storm.callback.WatcherCallBack;
+import org.apache.storm.nimbus.ILeaderElector;
+import org.apache.storm.nimbus.NimbusInfo;
+import org.apache.storm.utils.Utils;
+import org.apache.storm.utils.ZookeeperAuthInfo;
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.WatchedEvent;
+import org.apache.zookeeper.data.ACL;
+import org.apache.zookeeper.data.Stat;
+import org.apache.zookeeper.server.NIOServerCnxnFactory;
+import org.apache.zookeeper.server.ZooKeeperServer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.BindException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class Zookeeper {
+
+    private static Logger LOG = LoggerFactory.getLogger(Zookeeper.class);
+
+    public static CuratorFramework mkClient(Map conf, List<String> servers, Object port, String root) {
+        return mkClient(conf, servers, port, root, new DefaultWatcherCallBack());
+    }
+
+    public static CuratorFramework mkClient(Map conf, List<String> servers, Object port, Map authConf) {
+        return mkClient(conf, servers, port, "", new DefaultWatcherCallBack(), authConf);
+    }
+
+    public static CuratorFramework mkClient(Map conf, List<String> servers, Object port, String root, Map authConf) {
+        return mkClient(conf, servers, port, root, new DefaultWatcherCallBack(), authConf);
+    }
+
+    public static CuratorFramework mkClient(Map conf, List<String> servers, Object port, String root, final WatcherCallBack watcher, Map authConf) {
+        CuratorFramework fk;
+        if (authConf != null) {
+            fk = Utils.newCurator(conf, servers, port, root, new ZookeeperAuthInfo(authConf));
+        } else {
+            fk = Utils.newCurator(conf, servers, port, root);
+        }
+
+        fk.getCuratorListenable().addListener(new CuratorListener() {
+            @Override
+            public void eventReceived(CuratorFramework _fk, CuratorEvent e) throws Exception {
+                if (e.getType().equals(CuratorEventType.WATCHED)) {
+                    WatchedEvent event = e.getWatchedEvent();
+
+                    watcher.execute(event.getState(), event.getType(), event.getPath());
+                }
+
+            }
+        });
+        fk.start();
+        return fk;
+    }
+
+    /**
+     * connect ZK, register Watch/unhandle Watch
+     *
+     * @return
+     */
+    public static CuratorFramework mkClient(Map conf, List<String> servers, Object port, String root, final WatcherCallBack watcher) {
```

Comment:
Could you please file a follow up JIRA to make all of the 'Object port' parameters 'Integer port' instead?  This should be of the java Utils.newCurator too.